rollup.config.js

```diff
@@ -2,9 +2,57 @@ import resolve from '@rollup/plugin-node-resolve'
 import cjs from '@rollup/plugin-commonjs'
 import json from '@rollup/plugin-json'
 import ts from '@rollup/plugin-typescript'
+import alias from '@rollup/plugin-alias'
 
+// These extensions are not being used
 const extensions = ['.js', '.jsx', '.ts', '.tsx']
 
+function getEsbuildModulePath() {
+  /**
+   * The idea here is grasp Node require.resolve (which finds the dependency no matter
+   * where in the root the dependency is) to find `esbuild`.
+   *
+   * Although require.resolve returns the file specified in the `main` package.json, e.g.:
+   * /home/raulmelo/development/raulmelo-studio/node_modules/esbuild/lib/main.js
+   */
+  const fullPath = require.resolve('esbuild')
+  /**
+   * Since we just want the relative (til esbuild), we could simple filter everything
+   * AFTER the word "esbuild"
+   */
+  const regexEverythingAfterWord = /(?<=esbuild).*$/gm
+
+  /**
+   * and strip it out.
+   *
+   * Final result will be:
+   * /home/raulmelo/development/raulmelo-studio/node_modules/esbuild
+   */
+  return fullPath.replace(regexEverythingAfterWord, '')
+}
+
+/**
+ * This plugin injects a process env in build time.
+ *
+ * We could move the logic to determine the `ESBUILD_BINARY_PATH` at this point
+ * but it's totally up to you decide the best strategy.
+ */
+const injectPluginConfigured = injectProcessEnv({
+  ESBUILD_PATH: getEsbuildModulePath(),
+})
+
+/**
+ * And finally this approach solves duplicated react version.
+ * Here we use `alias` strategy combined with node require.resolve
+ * to the users versions of `react` and `react-dom`.
+ */
+const aliasPluginConfigured = alias({
```

Comment:
I just added this which ensure `next-mdx-prism` will use user's `react` instead some other react

Reply:
I didn't have success with this, although it might be some other issue:

```
Error: Invalid hook call. Hooks can only be called inside of the body of a function component. This could happen for one of the following reasons:
1. You might have mismatching versions of React and the renderer (such as React DOM)
2. You might be breaking the Rules of Hooks
3. You might have more than one copy of React in the same app
See https://fb.me/react-invalid-hook-call for tips about how to debug and fix this problem.
```